server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -712,7 +713,13 @@ public final String buildDescription() {
             sb.append("scroll[").append(scroll.keepAlive()).append("], ");
         }
         if (source != null) {
-            sb.append("source[").append(source.toString(FORMAT_PARAMS)).append("]");
+            sb.append("source[");
+            try {
+                sb.append(source.toString(FORMAT_PARAMS));
+            } catch (final OpenSearchException ex) {
```

Comment:
@reta / @dblock / @andrross : This is not right. buildDescription() shouldn't swallow exception. Anything unexpected should fail the request as early as possible. Earlier, IOException was wrapped and thrown and now that also gets swallowed. Why an object which is throwing UnsupportedOperationException not handled properly and reached till here at all? In order to fix an user behavior (which may not be an issue, as actual request may not reach till this point as well, unless someone has tested an actual user rest request and confirmed the behavior?). Tomorrow, somebody would wrap NPE also as OpenSearchException and start swallowing here as apparently buildDescription should succeed for some reason.

Reply:
>  Why an object which is throwing UnsupportedOperationException not handled properly and reached till here at all?

@shwetathareja There are few levels this exception will be raised and **it will be handled properly**, it is not swallowed.  The issue that the description of the request (serialization of the POJO) - the string representation - triggers the problem first, even before the request is processed (and validated). So the suggested fix here is to provide partial description of the request that will eventually fail for exactly the same cause. This is what tests are verifying.

Reply:
It is not an end-to-end test, The test simply testing a particular flow for ```GeoShapeQueryBuilder``` and expecting an exception.

Reply:
> Specifically in the request path the request is read, then traced (including its description, which throws exception), then validated & al.

@dblock : Request has to be parsed first/ deserialized. How can we trigger toContent() on object without creating the object?

Reply:
Ok that's fair, I think you know a lot more than me about the sequence of what's happening here and I don't have an actual example of when a linearring comes in as input (@andrross was asking for the same). 

So stepping back, I don't understand then why this code is present at all. If we can never submit a request with a linearring, should we delete all related code?

Reply:
Btw, I do think our fix of handling errors in `description` was right. We clearly have a case where `getDescription()` fails, and we always want a task to be able to describe itself.

Reply:
@dblock : I feel, the code we are fixing is actually dead code. Instead of changing exception handling here. +1 on cleaning up the dead code causing more confusion.

Reply:
> Will see if I can queue it, but I don't know the system well enough to be sure this is dead code though.

Same here, agree, not an expert on this.

Reply:
> The builder could validate the supported shapes in ` AbstractGeometryQueryBuilder` when creating the POJO itself.

I already replied to that here https://github.com/opensearch-project/OpenSearch/pull/12834#discussion_r1534520414

Reply:
@dblock : Are you reverting this change?

Reply:
No, I am still convinced that the implementation merged here is better than what we had. Today tasks can fail to `getDescription()` and the better place to guard it is in the code we added it in. If we get rid of `LinearRing`, then we don't need to protect from failed serialization.

Reply:
Once we remove the Linearring, the only exception, you are taking care in getDescription() is IOException which we shouldn't catch and swallow even for getDescription() to work. I don't think there is need for this exception handling considering it has always been like this, not impacting any other tests/ flow yet.

Reply:
I agree re:once we remove the Linearring. I opened https://github.com/opensearch-project/OpenSearch/issues/12910, I'll try to get to it.